src/pages/preferences/preferencesTabs/PublicKeysTab.tsx

```diff
@@ -1,3 +1,128 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { Table, Skeleton, Popconfirm } from "antd";
+import { useQuery, useMutation } from "@apollo/react-hooks";
+import Button from "@leafygreen-ui/button";
+import Icon from "@leafygreen-ui/icon";
+import styled from "@emotion/styled";
+import { GET_MY_PUBLIC_KEYS } from "gql/queries";
+import { useBannerDispatchContext } from "context/banners";
+import {
+  GetMyPublicKeysQuery,
+  GetMyPublicKeysQueryVariables,
+  RemovePublicKeyMutation,
+  RemovePublicKeyMutationVariables,
+} from "gql/generated/types";
+import { REMOVE_PUBLIC_KEY } from "gql/mutations";
 
-export const PublicKeysTab: React.FC = () => <div>Manage Public Keys</div>;
+export const PublicKeysTab: React.FC = () => {
+  const dispatchBanner = useBannerDispatchContext();
+  const [tableData, setTableData] = useState([]);
+  const { data: myKeysData, loading: loadingMyPublicKeys } = useQuery<
+    GetMyPublicKeysQuery,
+    GetMyPublicKeysQueryVariables
+  >(GET_MY_PUBLIC_KEYS, {
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error fetching your public keys: ${error.message}`
+      );
+    },
+  });
+  const [removePublicKey, { loading: loadingRemovePublicKey }] = useMutation<
+    RemovePublicKeyMutation,
+    RemovePublicKeyMutationVariables
+  >(REMOVE_PUBLIC_KEY, {
+    onCompleted(removeData) {
+      setTableData(removeData.removePublicKey);
+    },
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error removing the public key: ${error.message}`
+      );
+    },
+  });
+
+  useEffect(() => {
+    setTableData(myKeysData?.myPublicKeys ?? []);
+  }, [myKeysData]);
```

Comment:
why is this necessary?

Reply:
the payload from the myPublicKeys query and removePublicKey mutation represent a user's complete list public keys so I update the value in useState to render in the table.

Reply:
i think it could be better to use apollo client paradigms to update data after a mutation rather than using react state and hooks, which introduces a new paradigm of updating data after a mutation that we do not use anywhere else in the app. 

you could add a `refetchQueries` to the mutation, which will probably do just fine. or you could use the `update` function in the mutation to update the cache after the mutation succeeds, which will then update the value of `myKeysData` and rerender the table.

Reply:
wow, very cool. thanks for the idea!!